languagetool-core/src/main/java/org/languagetool/JLanguageTool.java

```diff
@@ -1201,7 +1214,8 @@ private int processColumnChange(int columnCount, String sentence) {
    */
   public List<RuleMatch> checkAnalyzedSentence(ParagraphHandling paraMode,
                                                List<Rule> rules, AnalyzedSentence analyzedSentence) throws IOException {
-    return checkAnalyzedSentence(paraMode, rules, analyzedSentence, false);
+    List<Rule> nonIgnored = rules.stream().filter(r -> !ignoreRule(r)).collect(Collectors.toList());
```

Comment:
Another unused method, public

Reply:
Also documented as "This is an internal method that's public only for technical reasons, please use one of the {@link #check(String)} methods instead"